internal/storage/bucket_handle_test.go

```diff
@@ -407,13 +408,99 @@ func (t *BucketHandleTest) TestListObjectMethodForMaxResult() {
+	// Validate that 2 objects are listed when MaxResults is passed 2.
 	AssertEq(nil, err2)
 	AssertEq(2, len(twoObj.Objects))
 	AssertEq(TestObjectRootFolderName, twoObj.Objects[0].Name)
 	AssertEq(TestObjectSubRootFolderName, twoObj.Objects[1].Name)
 	AssertEq(nil, twoObj.CollapsedRuns)
 }
 
+func (t *BucketHandleTest) TestListObjectMethodWithMissingMaxResult() {
+	// We have 4 objects in fakeserver.
+	fourObjWithMaxResults, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "",
+			Delimiter:                "",
+			IncludeTrailingDelimiter: true,
+			ContinuationToken:        "",
+			MaxResults:               100,
+			ProjectionVal:            0,
+		})
+
+	fourObjWithoutMaxResults, err2 := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "",
+			Delimiter:                "",
+			IncludeTrailingDelimiter: true,
+			ContinuationToken:        "",
+			//MaxResults:               2, // Missing MaxResults
+			ProjectionVal: 0,
+		})
+
+	// Validate that 4 objects are listed when MaxResults = 100, because there
+	// are total of 4 objects in fakebucket.
+	AssertEq(nil, err)
+	AssertEq(4, len(fourObjWithMaxResults.Objects))
+	AssertEq(TestObjectRootFolderName, fourObjWithMaxResults.Objects[0].Name)
+	AssertEq(TestObjectSubRootFolderName, fourObjWithMaxResults.Objects[1].Name)
+	AssertEq(TestSubObjectName, fourObjWithMaxResults.Objects[2].Name)
+	AssertEq(TestObjectName, fourObjWithMaxResults.Objects[3].Name)
+	AssertEq(nil, fourObjWithMaxResults.CollapsedRuns)
+
+	// Validate that all objects (4) are listed when MaxResults is not passed.
+	AssertEq(nil, err2)
+	AssertEq(4, len(fourObjWithoutMaxResults.Objects))
+	AssertEq(TestObjectRootFolderName, fourObjWithoutMaxResults.Objects[0].Name)
+	AssertEq(TestObjectSubRootFolderName, fourObjWithoutMaxResults.Objects[1].Name)
+	AssertEq(TestSubObjectName, fourObjWithoutMaxResults.Objects[2].Name)
+	AssertEq(TestObjectName, fourObjWithoutMaxResults.Objects[3].Name)
+	AssertEq(nil, fourObjWithoutMaxResults.CollapsedRuns)
+}
+
+func (t *BucketHandleTest) TestListObjectMethodWithZeroMaxResult() {
+	// We have 4 objects in fakeserver.
+	fourObj, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "",
+			Delimiter:                "",
+			IncludeTrailingDelimiter: true,
+			ContinuationToken:        "",
+			MaxResults:               100,
+			ProjectionVal:            0,
+		})
+
+	fourObjWithZeroMaxResults, err2 := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "",
+			Delimiter:                "",
+			IncludeTrailingDelimiter: true,
+			ContinuationToken:        "",
+			MaxResults:               0,
+			ProjectionVal:            0,
+		})
+
+	// Validate that 4 objects are listed when MaxResults = 100, because there
+	// are total of 4 objects in fakebucket.
+	AssertEq(nil, err)
```

Comment:
same here also

Reply:
Removed the first invocation and validation block and directly validating the test results with the expected data.